packages/dds/merge-tree/src/client.ts

```diff
@@ -597,6 +596,45 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 		);
 	}
 
+	/**
+	 * Returns a valid range for the op, or throws if the range is invalid
+	 * @param op - The op to generate the range for
+	 */
+	// eslint-disable-next-line import/no-deprecated
+	private getValidSidedRange(op: IMergeTreeObliterateSidedMsg | IMergeTreeObliterateMsg): {
+		start: InteriorSequencePlace;
+		end: InteriorSequencePlace;
+	} {
+		assert(op.pos1 !== undefined, "Expected start of obliterated range to be specified");
```

Comment:
we shouldn't use assert to validate user input, which this could be, we should throw user errors which will keep the exception text. 

it also doesn't appear that this function needs any member state of the client class, so i think we should move it to an unexported free function as those minimize better.

Reply:
I added checks for start and end being beyond the length of the sequence, which does require member state.